### PR TITLE
TRNG Plugin Triggers

### PR DESCRIPTION
**Pull request by Nickelony**
TODO:
- [x] Create new prj2 chunks for Triggers
- [x] Fix PR comments
- [x] Add plugin logic to .tr4 compiler
- [x] Fix TIDE not pointing new .prj2 files into Script directory
- [x] Add changes to Changes.txt
- [x] Expect script comments
- [x] Export trigger must have plugin ID added at the start: `$032000, 500, $3A` where `$03` is plugin ID from Script
- [x] Button lists should update selected Bits accordingly

### CODE REVIEW

File: TombLib/TombLib/NG/NgParameterInfo.cs [hunk 1]
```diff
@@ -269,7 +419,7 @@ public static ushort EncodeNGRealTimer(TriggerTargetType targetType, TriggerType
                             return timer(upperBound);
 
                         case TriggerTargetType.FlipEffect:
-                            NgTriggerSubtype flipEffectSubtriggerType = NgCatalog.FlipEffectTrigger.MainList.TryGetOrDefault(target);
+                            NgTriggerSubtype flipEffectSubtriggerType = NgCatalog.Instance.FlipEffectTrigger.MainList.TryGetOrDefault(target);
```
Comment (Nickelony) [inline TombLib/TombLib/NG/NgParameterInfo.cs]: Needs plugin condition

File: TombLib/TombLib/NG/NgParameterInfo.cs [hunk 2]
```diff
@@ -452,18 +604,18 @@ public static TriggerInstance ImportFromScriptTrigger(Level level, string script
                     case 0x9000:
                         result.TriggerType = TriggerType.ConditionNg;
                         result.TargetType = TriggerTargetType.ParameterNg;
-                        decode = !NgCatalog.ConditionTrigger.MainList[(ushort)(operands[2] & 0xFF)].Extra.IsEmpty;
+                        decode = !NgCatalog.Instance.ConditionTrigger.MainList[(ushort)(operands[2] & 0xFF)].Extra.IsEmpty;
```
Comment (Nickelony) [inline TombLib/TombLib/NG/NgParameterInfo.cs]: Needs plugin condition

**Comment by Jakub768** [comment]
with this PR, this means we can delete any occurrences of NGLE such as in TombIDE/TombIDE.Shared/TIDE/NGC

**Comment by Nickelony** [comment]
Besides some extra beta testing, this PR is complete!

**Pull request merged**